### PR TITLE
Upgrade to gulp4

### PR DESCRIPTION
**Pull request by mramato**
Supersedes #6992 which I tried to originally update but ran into some merge conflicts that just made it easier to re-do.

I did the simplest possible port for now since I didn't want to cram a whole bunch of clean-up into this PR.  Since we are still using `gulp.task` instead of the newer `module.exports` way of doing things, I had to re-order some of the tasks so that gulp.task can use them as a dependency.

Also updated `event-stream` and `karma` since they required no code changes on our end.

Thanks @SunBlack for getting this started!

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @mramato!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/gulp4/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: gulpfile.js [hunk 1]
```diff
@@ -544,13 +589,14 @@ function listAll(s3, bucketName, prefix, files, marker) {
...
 
-gulp.task('deploy-set-version', function() {
+gulp.task('deploy-set-version', function(done) {
     var buildVersion = yargs.argv.buildVersion;
     if (buildVersion) {
         // NPM versions can only contain alphanumeric and hyphen characters
         packageJson.version += '-' + buildVersion.replace(/[^[0-9A-Za-z-]/g, '');
         fs.writeFileSync('package.json', JSON.stringify(packageJson, undefined, 2));
     }
+    done();
```
Comment (hpinkos) [inline gulpfile.js]: Why does this need the `done` callback?  It doesn't look like anything is async

**Comment by hpinkos** [comment]
@mramato I just had that one question.  And CI failed.  Everything seems to be running fine locally

File: gulpfile.js (hunk 1 above)
Comment (mramato) [reply gulpfile.js]: gulp 4 dropped support for synchronous tasks, so you have to take a done callback (or return a stream or promise), you can't just "do nothing".  More details here: https://gulpjs.com/docs/en/getting-started/async-completion

**Comment by mramato** [comment]
This is ready (appveyor failures can be ignored until #7240 is merged)

File: gulpfile.js [hunk 2]
```diff
@@ -227,9 +266,15 @@ gulp.task('instrumentForCoverage', ['build'], function(done) {
         console.log(stdout);
         done();
     });
-});
+}));
+
+gulp.task('release', gulp.series('generateStubs', function() {
+    return combine()
```
Comment (hpinkos) [inline gulpfile.js]: Is there a reason we can't just pass all the functions to `gulp.series`?
ie `gulp.series('generateStubs', combine, minifyRelease, generateDocumentation)`

Comment (mramato) [reply gulpfile.js]: Good idea.  I just didn't notice it when changing the tasks. Done

**Comment by hpinkos** [comment]
Thanks @mramato!  This all looks good to me.  I'll merge as soon as CI passes

**Comment by hpinkos** [comment]
Cool, good enough for me

**Pull request merged**